### PR TITLE
Add intial documentation on `dev.eessi.io`

### PR DESCRIPTION
**Pull request by Neves-P**
This PR adds information on the `dev.eessi.io` CVMFS repository and basic usage wrt building and deploying software to `dev.eessi.io`.

This information will likely change in the future as the repo matures, but this should already provide a brief overview of the basic functionality.

### CODE REVIEW

File: docs/repositories/dev.eessi.io.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# `dev.eessi.io` repository
+
+## What is `dev.eessi.io`?
+
+`dev.eessi.io` is the development repository of EESSI. With it, developers can build and deploy non-production ready versions of their software to a CernVM-FS repository.
```
Comment (boegel) [inline docs/repositories/dev.eessi.io.md:5]:
```
`dev.eessi.io` is the development repository of EESSI. With it, developers can deploy pre-release builds of their software to EESSI. 
```

File: docs/repositories/dev.eessi.io.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# `dev.eessi.io` repository
+
+## What is `dev.eessi.io`?
+
+`dev.eessi.io` is the development repository of EESSI. With it, developers can build and deploy non-production ready versions of their software to a CernVM-FS repository. 
+This way, development version can easily be tested on systems where `dev.eessi.io` is available.
```
Comment (boegel) [inline docs/repositories/dev.eessi.io.md:6]:
```suggestion
This way, development versions of software can easily be tested on systems where the `dev.eessi.io` CernVM-FS repository is available.

File: docs/repositories/dev.eessi.io.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+On a system with `dev.eessi.io` mounted access is possible with `module use /cvmfs/dev.eessi.io/versions/2023.06/modules/all`. Then, all that is left is
```
Comment (boegel) [inline docs/repositories/dev.eessi.io.md:8]: This will change when we use subdirectories for projects

Comment (boegel) [reply docs/repositories/dev.eessi.io.md:8]: In fact, we probably (eventually) want a small custom init script for `dev.eessi.io` which:

* prints a warning (something like "`installations in `dev.eessi.io` are not for production use, only for testing`");
* sets up the environment for `software.eessi.io`;
* sets up the environment for a specific (set of) projects in `dev.eessi.io`;

File: docs/adding_software/adding_development_software.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# Adding software to `dev.eessi.io`
+
+## What is `dev.eessi.io`?
+
+`dev.eessi.io` is the development repository of EESSI. With it, developers can build and deploy non-production ready versions of their software to a CernVM-FS repository. This way, development version can easily be tested on systems where `dev.eessi.io` is available.
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:5]: This should just point to `repositories/dev.eessi.io.md`

File: docs/adding_software/adding_development_software.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+## Adding software
+
+Using `dev.eessi.io` is similar to using EESSI's production repository `software.eessi.io`. Software builds are triggered by a bot listening to pull requests in GitHub repository (at the moment https://github.com/EESSI/dev.eessi.io). This repository is organised by project, where corresponding easystack files and easyconfig files are placed.
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:9]: We'll probably ends up with separate repos per project, so point to https://github.com/search?q=org%3AEESSI+dev.eessi.io&type=repositories ?

File: docs/adding_software/adding_development_software.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+```
+dev.eessi.io
+├── project1
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:13]: Not correct I think, each repo has it's own separate file structure.

Maybe use `dev.eessi.io-example` here (we could even create that repo once things have stabilized a bit)

File: docs/adding_software/adding_development_software.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+│   ├── easyconfigs
+│   └── easystacks
+└── project2
+    ├── easyconfigs
+    └── easystacks
+```
+
+Creating a PR that adds an entry to an easystack file under on of the projects will allow authorised users to trigger
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:21]:
```suggestion
Creating a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) that adds an entry to an easystack file under one of the projects will allow authorized users to trigger

File: docs/adding_software/adding_development_software.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+├── project1
+│   ├── easyconfigs
+│   └── easystacks
+└── project2
+    ├── easyconfigs
+    └── easystacks
+```
+
+Creating a PR that adds an entry to an easystack file under on of the projects will allow authorised users to trigger 
+a build with the command `bot: build` through a GitHub comment.
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:22]: refer to bot docs here?

File: docs/adding_software/adding_development_software.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+## Building development versions
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:24]:
```suggestion
Deploying pre-release builds of software
```

I would try and make this a clear 5-step (or however many steps) plan.

File: docs/adding_software/adding_development_software.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+To allow for development builds, we leverage the `--software-commit` functionality (requires EasyBuild v4.9.3 or higher). This lets us build a given application from
+a specific commit in repository. This can also be done from a fork, by changing the `github_account` field in the easyconfig file. 
+We've created a template for `ESPResSo` based on the standard eaasyconfig of the most recent version. The relevant fields are:
+
+``` python
+easyblock = 'CMakeMake'
+
+name = 'ESPResSo'
+version = '4.2.2'
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:39]: use `devel` here?

File: docs/adding_software/adding_development_software.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+versionsuffix = '-%(software_commit)s'
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:40]: Random thought: I would be brilliant if we could add support for something like `%(software_commit_datestamp)s`, which is automatically replaced with the datestamp of the commit (so you get module names that sort well):
```python
name = 'ESPResSo'
version = '%(software_commit_datestamp)s-%(software_commit)s'
```

File: docs/adding_software/adding_development_software.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+description = """A software package for performing and analyzing scientific Molecular Dynamics simulations."""
+
+github_account = 'espressomd'
+source_urls = ['https://github.com/%(github_account)s/%(name)s/archive/']
+
+sources = ['%(software_commit)s.tar.gz']
+```
+
+One can also make new changes to the easyconfig file, for example, if the new functionality requires new build or 
+runtime dependencies, patches, flags, etc. It's a good idea to try installing from a specific commit locally first,
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:52]:
```suggestion
runtime dependencies, patches, configuration options, etc. It's a good idea to try installing from a specific commit locally first,

File: docs/adding_software/adding_development_software.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+to at least see if everything is parsed correctly and confirm that the right sources are being downloaded.
+
+### Easystack files and triggering builds
+
+After the easyconfig file has been created and added to `projectX/easyconfigs`, an easystack file that picks it up
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:57]:
```suggestion
After the easyconfig file has been created and added to the `easyconfigs` subdirectory, an [easystack file](https://docs.easybuild.io/easystack-files) that picks it up

File: docs/adding_software/adding_development_software.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+needs to be in place. This easystack file must follow a naming convention: `software-eb-X.Y.Z-dev.yml`,
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:58]: Make the part about the filename a note that pops out a bit (see https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage)

File: docs/adding_software/adding_development_software.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+Following our example for `ESPREsSo`, it would look like: 
+
+``` yml
+easyconfigs:
+  - ESPResSo-4.2.2-foss-2023a-software-commit.eb:
+      options:
+        software-commit: 2ba17de6096933275abec0550981d9122e4e5f28 # release 4.2.2
+```
+
+The `ESPResSo-4.2.2-foss-2023a-software-commit.eb` would be the easyconfig file added in the step above.
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:69]: rename to `ESPResSo-devel-foss-2023a.eb` ?

File: docs/adding_software/adding_development_software.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+  - ESPResSo-4.2.2-foss-2023a-software-commit.eb:
+      options:
+        software-commit: 2ba17de6096933275abec0550981d9122e4e5f28 # release 4.2.2
+```
+
+The `ESPResSo-4.2.2-foss-2023a-software-commit.eb` would be the easyconfig file added in the step above. 
+Note the option passing the `software-commit` for the development version that should be built. 
+For the sake of this example, the chosen commit actually corresponds to the 4.2.2 release.
+
+To trigger a build, all one needs to do is open a PR with the changes adding the easyconfig and easystack
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:73]: Again, refer to bot docs

File: docs/adding_software/adding_development_software.md [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+Note the option passing the `software-commit` for the development version that should be built. 
+For the sake of this example, the chosen commit actually corresponds to the 4.2.2 release.
+
+To trigger a build, all one needs to do is open a PR with the changes adding the easyconfig and easystack 
+files and commenting `bot: build`. This can only be done by previously authorized users. 
+The current build cluster builds for the `zen2` CPU microarchitecture, but this is likely to change.
+
+Once a build is complete and the `bot:deploy` label is added, a staging PR can be merged to deploy the
+application to the `dev.eessi.io` cvmfs repository. On a system with `dev.eessi.io` mounted, then all
+that is left is to `module use /cvmfs/dev.eessi.io/versions/2023.06/modules/all` and try out the software!
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:79]: `module use` needs project subdir

**Review by boegel — CHANGES_REQUESTED** [review]

File: docs/adding_software/adding_development_software.md (hunk 9 above)
Comment (Neves-P) [reply docs/adding_software/adding_development_software.md:24]: Good point, I agree

File: docs/repositories/dev.eessi.io.md (hunk 1 above)
Comment (Neves-P) [reply docs/repositories/dev.eessi.io.md]: Addressed in c6b3ac0]

File: docs/adding_software/adding_development_software.md (hunk 9 above)
Comment (Neves-P) [reply docs/adding_software/adding_development_software.md]: Done in b8517b6

File: docs/adding_software/adding_development_software.md (hunk 15 above)
Comment (Neves-P) [reply docs/adding_software/adding_development_software.md]: Done in c1ed77be

File: docs/adding_software/adding_development_software.md (hunk 14 above)
Comment (Neves-P) [reply docs/adding_software/adding_development_software.md]: Done in 964ab5b9

File: docs/adding_software/adding_development_software.md (hunk 16 above)
Comment (Neves-P) [reply docs/adding_software/adding_development_software.md]: Done in b8517b6

File: docs/adding_software/adding_development_software.md [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+While the process to build for `dev.eessi.io` is similar to the one for the [production repository](../repositories/software.eessi.io.md) there 
+are a few additional details to keep in mind.
+
+#### Versions
+
+Installations to the EESSI production repository refer to specific versions of applications, however, Development builds can't follow the same 
+approach as they are most often not pegged to a release. Because of this, it is possible to add a descriptive "version" name to the `version` field
+in the easyconfig file for a given (set of) installations. 
+
+Note that some applications are built with custom easyblocks which may
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:91]:
```suggestion
Note that some applications are built with custom easyblocks, which may

File: docs/adding_software/adding_development_software.md [hunk 19]
```diff
@@ -0,0 +1,155 @@
...
+use the `version` field to determine how the installation is meant to work (e.g., versions >2 need to copy files from to a new directory). Make sure
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:92]:
```suggestion
use the `version` parameter to determine how the installation is meant to work (for example, recent versions need to copy files from to a new directory). Make sure

File: docs/adding_software/adding_development_software.md [hunk 20]
```diff
@@ -0,0 +1,155 @@
...
+
+One can also make new changes to the easyconfig file, for example, if the new functionality requires new build or 
+runtime dependencies, patches, configuration options, etc. It's a good idea to try installing from a specific commit locally first,
+to at least see if everything is parsed correctly and confirm that the right sources are being downloaded.
+
+
+While the process to build for `dev.eessi.io` is similar to the one for the [production repository](../repositories/software.eessi.io.md) there 
+are a few additional details to keep in mind.
+
+#### Versions
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:85]:
```suggestion
#### Software version

File: docs/adding_software/adding_development_software.md [hunk 21]
```diff
@@ -0,0 +1,155 @@
...
+Note that some applications are built with custom easyblocks which may
+use the `version` field to determine how the installation is meant to work (e.g., versions >2 need to copy files from to a new directory). Make sure
+that you account for this, otherwise you may install software differently than intended. If you encounter issues, you can open an issue in our
+[support portal](https://gitlab.com/eessi/support#eessi-support-portal).
+
+#### Installing dependencies
+
+Installations in `dev.eessi.io` are done _on top_ of `software.eessi.io`. That means if your development build depends on some application that is
+already installed in `software.eessi.io`, then that will simply be used. However, if you need to add a new dependency, then this must included as 
+part of the build. That means including an easystack file for it, and adding it to the right easystack file.
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:100]:
```suggestion
part of the build. That means including an easyconfig file for it, and adding it to the right easystack file.

File: docs/adding_software/adding_development_software.md [hunk 22]
```diff
@@ -0,0 +1,155 @@
...
+
+#### Using commit IDs or tags for `--software-commit`
+
+Installing with `--software-commit` requires that you include either a commit ID or a tag. The installation procedure will use the commit ID or tag to
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:104]:
```suggestion
Installing with `--software-commit` requires that you include either a commit ID or a tag. The installation procedure will use this to

File: docs/adding_software/adding_development_software.md [hunk 23]
```diff
@@ -0,0 +1,155 @@
...
+#### Installing dependencies
+
+Installations in `dev.eessi.io` are done _on top_ of `software.eessi.io`. That means if your development build depends on some application that is
+already installed in `software.eessi.io`, then that will simply be used. However, if you need to add a new dependency, then this must included as 
+part of the build. That means including an easystack file for it, and adding it to the right easystack file.
+
+#### Using commit IDs or tags for `--software-commit`
+
+Installing with `--software-commit` requires that you include either a commit ID or a tag. The installation procedure will use the commit ID or tag to 
+obtain the source from the right state. Because tags can be changed to point to a different commit ID, we recommend you avoid using them but stick with
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:105]:
```suggestion
obtain the sources for the build. Because tags can be changed to point to a different commit ID, we recommend you avoid using them and sticking to

File: docs/adding_software/adding_development_software.md [hunk 24]
```diff
@@ -0,0 +1,155 @@
...
+the commit ID itself. You can then include this in the `versionsuffix` on your easyconfig file, to generate a unique (if ugly) module name.
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:106]:
```suggestion
the commit ID itself. You can then include this in the `versionsuffix` on your easyconfig file, to generate a unique (though "ugly") module name.

File: docs/adding_software/adding_development_software.md [hunk 25]
```diff
@@ -0,0 +1,155 @@
...
+needs to be in place so that a build can be triggered.
+
+!!! note "Naming convention for easystack files"
+
+    The easystack files must follow a naming convention and be named something
+    like: `software-eb-X.Y.Z-dev.yml`, where X.Y.Z correspond to the EasyBuild
+    version used to install the software. Following our example for 
+    `ESPREsSo`, it would look like: 
+    
+    ``` yml
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:133]:
````suggestion
    ```yaml

File: docs/adding_software/adding_development_software.md [hunk 26]
```diff
@@ -0,0 +1,155 @@
...
+    easyconfigs:
+      - ESPResSo-devel-foss-2023a-software-commit.eb:
+          options:
+            software-commit: 2ba17de6096933275abec0550981d9122e4e5f28 # release 4.2.2
+    ```
+
+The `ESPResSo-devel-foss-2023a-software-commit.eb` would be the easyconfig file added in our example step above.
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:140]:
```suggestion
`ESPResSo-devel-foss-2023a-software-commit.eb` would be the name of the easyconfig file added in our example step above.

File: docs/adding_software/adding_development_software.md [hunk 27]
```diff
@@ -0,0 +1,155 @@
...
+    ``` yml
+    easyconfigs:
+      - ESPResSo-devel-foss-2023a-software-commit.eb:
+          options:
+            software-commit: 2ba17de6096933275abec0550981d9122e4e5f28 # release 4.2.2
+    ```
+
+The `ESPResSo-devel-foss-2023a-software-commit.eb` would be the easyconfig file added in our example step above. 
+Note the option passing the `software-commit` for the development version that should be built.
+For the sake of this example, the chosen commit actually corresponds to the 4.2.2 release.
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:142]:
```suggestion
For the sake of this example, the chosen commit actually corresponds to the [4.2.2 release of ESPResSo](https://github.com/espressomd/espresso/releases/tag/4.2.2).

File: docs/adding_software/adding_development_software.md [hunk 28]
```diff
@@ -0,0 +1,155 @@
...
+
+### Triggering builds
+
+We use the [EESSI build-test-deploy bot](../bot.md) to handle software builds.
+All one needs to do is open a PR with the changes adding the easyconfig and easystack 
+files and commenting `bot: build`. This can only be done by previously authorized users. 
+The current build cluster builds only for the `zen2` CPU microarchitecture, but this is likely to change.
```
Comment (boegel) [inline docs/adding_software/adding_development_software.md:149]:
```suggestion
The current build cluster for `dev.eessi.io` builds only for the `zen2` CPU microarchitecture, but this is likely to change.
```

File: docs/adding_software/adding_development_software.md (hunk 17 above)
Comment (boegel) [reply docs/adding_software/adding_development_software.md:153]: OK, let's not wait for the project subdirectories

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]
